contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -1424,36 +1424,63 @@ $BODY$
 LANGUAGE plpgsql
 IMMUTABLE;
 
--- Duplicate functions with arg TEXT since ANYELEMNT cannot handle type unknown.
-CREATE OR REPLACE FUNCTION sys.stuff(expr TEXT, start INTEGER, length INTEGER, replace_expr TEXT)
-RETURNS TEXT AS
+-- wrapper functions for stuff --
+CREATE OR REPLACE FUNCTION sys.stuff(expr sys.VARBINARY, start INTEGER, length INTEGER, replace_expr sys.VARCHAR)
+RETURNS VARBINARY
+AS
 $BODY$
-SELECT
-CASE
-WHEN start <= 0 or start > length(expr) or length < 0 THEN
-	NULL
-WHEN replace_expr is NULL THEN
-	overlay (expr placing '' from start for length)
-ELSE
-	overlay (expr placing replace_expr from start for length)
+BEGIN
+    IF start IS NULL OR expr IS NULL OR length IS NULL THEN
+        RETURN NULL;
+    END IF;
+    IF start <= 0 OR start > sys.len(expr) OR length < 0 THEN
+        RETURN NULL;
+    END IF;
+    IF replace_expr IS NULL THEN
+        RETURN (SELECT (overlay (expr::sys.VARCHAR placing '' from start for length))::sys.VARCHAR)::VARBINARY;
```

Comment:
Let's be consistent with the use of `VARBINARY` and `sys.VARBINARY`. We should should use only one i.e. `sys.VARBINARY`.

Reply:
Done